annotation-solr/src/main/java/eu/europeana/annotation/solr/service/impl/SolrAnnotationServiceImpl.java

```diff
@@ -589,6 +592,35 @@ private String solrUniquenessQueryLinking(Annotation anno) {
 		solrQuery+=")";
 		return solrQuery;
 	}
+	
+	private String solrUniquenessQueryLinkForContributing(Annotation anno) {
+      String solrQuery = "(" + WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.LINKFORCONTRIBUTING.getOaType() + "\"";
+      
+      List<String> targetValues = anno.getTarget().getValues();
+      if(targetValues!=null) {
```

Comment:
this type of annotation has only one target URI. If there are nore target URIs we need to throw and exception

Reply:
Improved. The check for this is in the validator class, i.e. in this method eu.europeana.annotation.web.service.impl.BaseAnnotationValidator.